src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts

```diff
@@ -432,25 +445,48 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
         this.quizExerciseService.findForCourse(selectedCourse.id!).subscribe(
             (quizExercisesResponse: HttpResponse<QuizExercise[]>) => {
                 if (quizExercisesResponse.body) {
-                    const quizExercises = quizExercisesResponse.body!;
-                    for (const quizExercise of quizExercises) {
-                        this.quizExerciseService.find(quizExercise.id!).subscribe((response: HttpResponse<QuizExercise>) => {
-                            const quizExerciseResponse = response.body!;
-                            if (quizExerciseResponse.quizQuestions && quizExerciseResponse.quizQuestions.length > 0) {
-                                for (const question of quizExerciseResponse.quizQuestions) {
-                                    question.exercise = quizExercise;
-                                    this.allExistingQuestions.push(question);
-                                }
-                            }
-                            this.applyFilter();
-                        });
-                    }
+                    this.applyQuestionsAndFilter(quizExercisesResponse.body!);
+                }
+            },
+            (res: HttpErrorResponse) => this.onError(res),
+        );
+    }
+
+    onExamSelect(): void {
+        this.allExistingQuestions = this.existingQuestions = [];
+        if (!this.selectedExamId) {
+            return;
+        }
+
+        /** Search the selected exam by id in all available exams **/
+        const selectedExam = this.exams.find((exam) => exam.id === Number(this.selectedExamId))!;
+
+        // For the given exam, get list of all quiz exercises. And for all quiz exercises, get list of all questions in a quiz exercise
+        this.quizExerciseService.findForExam(selectedExam.id!).subscribe(
+            (quizExercisesResponse: HttpResponse<QuizExercise[]>) => {
+                if (quizExercisesResponse.body) {
+                    this.applyQuestionsAndFilter(quizExercisesResponse.body!);
                 }
             },
             (res: HttpErrorResponse) => this.onError(res),
         );
     }
 
+    private applyQuestionsAndFilter(quizExercises: QuizExercise[]) {
+        for (const quizExercise of quizExercises) {
+            this.quizExerciseService.find(quizExercise.id!).subscribe((response: HttpResponse<QuizExercise>) => {
+                const quizExerciseResponse = response.body!;
+                if (quizExerciseResponse.quizQuestions && quizExerciseResponse.quizQuestions.length > 0) {
+                    for (const question of quizExerciseResponse.quizQuestions) {
+                        question.exercise = quizExercise;
+                        this.allExistingQuestions.push(question);
+                    }
+                }
+                this.applyFilter();
+            });
+        }
+    }
```

Comment:
wouldn't it be better to call `this.applyFilter()` after the for-loop?

Reply:
No, already tried that. No quizQuestions are shown anymore when moving the applyFilter() to after the for-loop